File: ReflexGame/app/src/main/java/com/example/reflexgame/MainActivity.java
package com.example.reflexgame;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.os.Handler;
import android.widget.Button;
import android.widget.RelativeLayout;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    public Button button1, button2;
    public RelativeLayout relativeLayout;

    Runnable runnable = new Runnable() {
        @SuppressLint("ResourceAsColor")
        @Override
        public void run() {
            relativeLayout.setBackgroundColor(R.color.green);

            final long time = System.currentTimeMillis();

            button2.setOnClickListener(view -> {
                long time1  = System.currentTimeMillis();
                Toast.makeText(getApplicationContext(), "Your reflexes takes " + (time1 - time) + " time to work", Toast.LENGTH_LONG ).show();
                relativeLayout.setBackgroundColor(0);
            });
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        relativeLayout = findViewById(R.id.rlVarl);
        button1 = findViewById(R.id.btVar1);
        button2 = findViewById(R.id.btVar2);

        button1.setOnClickListener(view -> {

            Random random = new Random();
            int num = random.nextInt(10);

            Handler handler = new Handler();
            handler.postDelayed(runnable, num * 1000);
        });
    }
}

File: PalindromeChecker/app/src/main/java/com/example/palindromechecker/MainActivity.kt
package com.example.palindromechecker

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val buttonCheck =findViewById<Button>(R.id.ButtonCheck)
        val editText = findViewById<EditText>(R.id.editText)
        buttonCheck.setOnClickListener {
            val text = editText.text.toString()

            if (ispalindrome(text)) {
                Toast.makeText(this, "Entered word is palindrome",Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "Entered word is not a Palindrome",Toast.LENGTH_SHORT).show()
            }

        }

    }

    private fun ispalindrome(text: String): Boolean {
        val reverseString = text.reversed().toString()
        return text.equals(reverseString, ignoreCase= true)
    }
}